### PR TITLE
Fixed: CF score bypassing upgrades not being allowed if a revision upgrade is present

### LINKED ISSUE #10751: Pull Sonarr commit 'Fixed: Custom Format score bypassing upgrades not being allowed'
Pull https://github.com/Sonarr/Sonarr/commit/ebe23104d4b29a3c900a982fb84e75c27ed531ab

```
git cherry-pick -ex ebe23104d4b29a3c900a982fb84e75c27ed531ab
```

This commit has 5 conflict(s)

You can ask me to `close all`, `open pr here` or `open pr everywhere`

### PR DESCRIPTION
**Pull request by CeruleanRed**
#### Description
Similar to Sonarr's [issue](https://github.com/Sonarr/Sonarr/pull/7460) of `Upgrades allowed` being bypassed if there was a revision upgrade. Fixed in the same way as the Sonarr version.

- Closes #10751

### CODE REVIEW

**Comment by CeruleanRed** [comment]
Sorry, I'm a bit confused by the process. Should I ask the bot to create another PR and resolve the merge issues there or wait until the `Don't merge` flag gets removed?

**Comment by CeruleanRed** [comment]
Sorry, it didn't actually cross my mind that I could cherry-pick from a different repo. I think it should be correct now.

**Review by mynameisbogdan — APPROVED** [review]

**Pull request merged**
